internal/controllers/topology/cluster/cluster_controller_test.go

```diff
@@ -1067,3 +1069,245 @@ func referenceExistsWithCorrectKindAndAPIVersion(reference *corev1.ObjectReferen
 	}
 	return nil
 }
+
+func TestReconciler_DefaultCluster(t *testing.T) {
```

Comment:
These defaulting and validation tests are close to smoke tests - just ensuring we're running these in the reconciler.